src/module/hook-multi-plexer/usage.ts

```diff
@@ -0,0 +1,120 @@
+import { Address, encodeFunctionData, Hex, PublicClient } from 'viem'
+import { HOOK_MULTI_PLEXER_ADDRESS } from './constants'
+import { abi } from './abi'
+import { Execution } from '../../account'
+import { Account } from '../../account'
+import { HookType } from './types'
+
+export const getHooks = async ({
+  account,
+  client,
+}: {
+  account: Account
+  client: PublicClient
+}): Promise<Address[]> => {
+  try {
+    const hooks = (await client.readContract({
+      address: HOOK_MULTI_PLEXER_ADDRESS,
+      abi,
+      functionName: 'getHooks',
+      args: [account.address],
+    })) as Address[]
+
+    return hooks
+  } catch (err) {
+    console.error(err)
+    return []
+  }
+}
+
+export const addHook = ({
+  hook,
+  hookType,
+}: {
+  hook: Address
+  hookType: HookType
+}): Execution => {
+  try {
+    return {
+      target: HOOK_MULTI_PLEXER_ADDRESS,
+      value: BigInt(0),
+      callData: encodeFunctionData({
+        functionName: 'addHook',
+        abi,
+        args: [hook, hookType],
+      }),
+    }
+  } catch {
+    throw new Error(`Failed to add hook ${hook}`)
+  }
+}
+
+export const addSigHook = ({
+  hook,
+  sig,
+  hookType,
```

Comment:
could it make sense to collapse these into a single func with an optional sig param? this is annoying in solidity hence isnt the case in the contract but could work here right?

Reply:
@kopy-kat Yeah make sense, will do with the remove hook as well